### ISSUE TITLE
Explain only part of generated tokens

### ISSUE THREAD

**Issue by lizc126** [rachtibat/LRP-for-Transformers#39]
Hi! Very insteresting work and thanks a lot for putting them so nicely! 

In the examples, the evaluated token is only the last (and the only) new token, I am curious how to extend this to  more tokens? 
For example, my model would generate a json struction like ```{"venue": "International Conference on Machine Learning"}```. I want to use `International Conference on Machine Learning` as a reference to look into my prompt. But in each forward pass like `output_logits = model(inputs_embeds=input_embeds.requires_grad_(), use_cache=False).logits`, only one token is generated at each step. 

My idea is to run it multiple times starting from ```<my prompt> + {"venue": ```, but then the most important tokens would always be the tokens before the generated one. Do you have any ideas how to achieve this?

Thank you!

**Comment by rachtibat** [comment]
Hi @lizc126,

you can achieve this very easily! Just generate the entire sequence with cache. Save the generated ids.
Then do 1 forward pass with entire sequence and one backward pass, where you do a backward pass on all tokens you want to explain, e.g. `output_logits[:, indices, generated_ids].backward(torch.ones_like(output_logits[:, indices, generated_ids]))`

Best

**Comment by rachtibat** [comment]
Note, I edited the code, there was a part missing

**Comment by lizc126** [comment]
thank you for the help, it's working now!
